packages/grid/src/elements/Col.tsx

```diff
@@ -7,38 +7,91 @@
 
 import React, { HTMLAttributes } from 'react';
 import PropTypes from 'prop-types';
+import { TYPE_ALIGN_SELF, TYPE_NUMBER, ARRAY_ALIGN_SELF } from '../utils/types';
 import { StyledCol } from '../styled';
+import useGridContext from '../utils/useGridContext';
 
 export interface IColProps extends HTMLAttributes<HTMLDivElement> {
-  /** Sizing for all breakpoints. */
-  size?: number | string;
-  /** Sizing for extra small breakpoints. */
-  xs?: number | string | boolean;
-  /** Sizing for small breakpoints. */
-  sm?: number | string | boolean;
-  /** Sizing for medium breakpoints. */
-  md?: number | string | boolean;
-  /** Sizing for large breakpoints. */
-  lg?: number | string | boolean;
-  /** Sizing for extra large breakpoints. */
-  xl?: number | string | boolean;
-  offsetXs?: number | string | boolean;
-  offsetSm?: number | string | boolean;
-  offsetMd?: number | string | boolean;
-  offsetLg?: number | string | boolean;
-  offsetXl?: number | string | boolean;
-  /** Use flexbox alignment utilities to horizontally align */
-  alignSelf?: 'start' | 'center' | 'end';
-  justifyContent?: 'start' | 'center' | 'end' | 'around' | 'between';
-  order?: any;
+  /**
+   * Determine the size, relative to the total number of `columns` in the grid,
+   * for all screen sizes
+   */
+  size?: TYPE_NUMBER;
+  /** Determine the size for extra-small screen sizes */
+  xs?: TYPE_NUMBER | boolean;
```

Comment:
If the sizing props have a consistent type of `string | number | boolean` would it make sense to create another type for this?

Reply:
I thought about adding a `TYPE_SIZE` for these. But then it's weird that `size` would not be `TYPE_SIZE`. So then I thought about adding a `TYPE_NUMBER_OR_BOOLEAN`, but that reads worse than just saying `TYPE_NUMBER | boolean`.

Reply:
Could we define 2 additional types here for the `xs|sm|etc.` and `offset*` props? This would allow:

```tsx
xs: TYPE_COLUMN,
offsetXs: TYPE_COLUMN_OFFSET
```

Even thought they allow roughly the same basic types `string | number | (sometimes) boolean` these values aren't necessarily related.

Reply:
Added `BREAKPOINT` constant to cover `number | string | boolean` types.